src/components/Forms/Pledge/GeneralPledge.js

```diff
@@ -67,21 +69,24 @@ export default ({ pledgeId }) => {
 
   return (
     <Form
-      onSubmit={e => {
+      onSubmit={async e => {
         e.pledgeId = pledgeId;
         e.privacyConsent = true;
         e.newsletterConsent = true;
         setHasSubmitted(true);
         setPledgeLocally(e);
         if (!isAuthenticated) {
-          signUp(e.email);
+          await signUp(e.email);
         }
       }}
       initialValues={{
         name: userId !== undefined && userData ? userData.username : '',
         zipCode: userId !== undefined && userData ? userData.zipCode : '',
       }}
       validate={values => validate(values, isAuthenticated)}
+      initialValues={{
```

Comment:
Small thing: this is a duplicate prop. There's another initial values prop in use on line 82.